pkg/operator/k8sutil/placement.go

```diff
@@ -36,6 +38,19 @@ func (p Placement) ApplyToPodSpec(t *v1.PodSpec) {
 		}
 		t.Affinity.NodeAffinity = p.NodeAffinity
 	}
+	if p.PodAffinity != nil {
+		if t.Affinity == nil {
```

Comment:
If it's wanted we can deduplicate this code by a single if condition before the conditions begin (line 34/5).
Something like this, before line 35:
```
if p.NodeAffinity != nil || p.PodAntiAffinity != nil || p.PodAntiAffinity != nil  {
    t.Affinity = &v1.Affinity{}
}
```

Reply:
what if we always initialize the Affinity to empty? Is there a side effect when no affinity settings are intended?

Reply:
When updating "old" replicasets they may get updated due to the empty affinity, but I think that Kubernetes is smart enough to see that the change is empty and do nothing.